lightning/src/routing/network_graph.rs

```diff
@@ -1737,8 +1821,7 @@ mod tests {
 
 	}
 
-	#[test]
-	fn handling_network_update() {
+	fn do_handling_network_update(remove_by_timeout: bool) {
```

Comment:
Why use this test? It's used to test handling of `NetworkUpdate` from `PaymentPathFailed` events, which is unrelated to this.

Reply:
Because I didn't want to sit down and write out an entirely redundant test which copies all the manual message creation stuff. I could, I suppose, but its a bunch of additional test code to do the same thing over and over, which tends to create pain later.

Reply:
Wouldn't the `handling_channel_update` test be a better? I'd prefer to not test something unrelated in `handling_network_update`.

Reply:
Wouldn't the `handling_channel_update` test be a better place? It's already testing timestamp-related code.

Reply:
Hmm, that handles rejecting some channel updates (which ideally we'd test the new "reject if in the past or future" checks in, but sadly for now we have to disable those in testing as ~every one of our tests currently fails with it :/), but `network_update` here tests removing things which felt like it lined up better? I dunno, if you feel strongly I can move it.

Reply:
Hmm... could we just make a new test and copy the setup part, then? That part really isn't much code and could always be refactored later. I just have a preference for keeping tests limited as it makes reading them easier than if they have conditional logic about what is being tested.

Reply:
Ok, I refactored the tests in network_graph top-to-bottom so that adding new tests doesn't result in nearly as much code duplication then moved the new test code into a new tests.